changelog/0087-upgrade-to-es-6.yml

```
@@ -0,0 +1,405 @@
+    ```sh
+      cchq ${ENV} run-shell-command elasticsearch  "du -sh  /opt/data/elasticsearch-5.6.16" -b
+    ```
+
+    5. After you have figured out the backup strategy and ensured sufficient disk space, continue with the steps below.
+  
+  ### Upgrade Steps
+
+  1. Add the following parameters to your environment's `public.yml`:
+    
+    ```yaml
+    elasticsearch_version: 6.8.23
+    elasticsearch_download_sha256: 424af91f838f9e5f13e0292f97cbd6333535450291a621d761bd479dfc2dff78.
+    .
+    .
+    localsettings:
+      ...
+      ELASTICSEARCH_MAJOR_VERSION: 6
+    ```
+
+  2. Pull the latest commcare-cloud changes on control machine.
+    
+    ```sh
+      cd ${COMMCARE_CLOUD_REPO} && git pull
+    ```
+
+  3. Stop commcare services.
+
+    ```sh
+      cchq ${ENV} downtime start
+    ```
+  
+  4. Apply the settings that you have updated in Step 1.
+
+    ```sh
+      cchq ${ENV} update-config
+    ```
+
+  5. Disable Elasticsearch shard allocation. It should already be disabled but for ensuring that its already disabled.
+
+    ```sh
+      curl -X PUT "${ES_HOST}:9200/_cluster/settings?pretty" -H 'Content-Type: application/json' -d'
+      {
+        "persistent": {
+          "cluster.routing.allocation.enable": "none"
+        }
+      }
+      '
+
+    ```
+
+  6. Perform a synced flush.
+
+    ```sh
+      curl -X POST "${ES_HOST}:9200/_flush/synced?pretty"  # safe to reissue a few times if it fails
+    ```
+
+  7. Stop the Monit and Elasticsearch services.
+
+    ```sh
+      cchq ${ENV} run-shell-command elasticsearch "service monit stop" -b
+
+      cchq ${ENV} run-shell-command elasticsearch "service elasticsearch stop" -b
+    ```
+
+  8. Make a backup of the 5.6.16 data directory. If you are backing up on the same machine then one way to do it is by:
+    
+    ```sh
+      cchq ${ENV} run-shell-command elasticsearch "cp -r /opt/data/elasticsearch-5.6.16 /opt/data/elasticsearch-5.6.16-backup" -b
+    ```
+  9. Install and run the new version of Elasticsearch.
+
+    ```sh
+      cchq ${ENV} ansible-playbook deploy_stack.yml --limit=elasticsearch --tags=elasticsearch
+    ```
+
+  10. Stop the Monit and Elasticsearch services.
+
+    ```sh
+      cchq ${ENV} run-shell-command elasticsearch "service monit stop" -b
+
+      cchq ${ENV} run-shell-command elasticsearch "service elasticsearch stop" -b
+    ```
+
+  11. Rename the newly created 6.8.23 data directory so it’s out of the way and then move the 5.6.16 data directory to the new 6.8.23 data directory location.
+
+    ```sh
+      cchq ${ENV} run-shell-command elasticsearch "mv /opt/data/elasticsearch-6.8.23 /opt/data/elasticsearch-6.8.23-new-installation" -b
+
+      cchq ${ENV} run-shell-command elasticsearch "mv /opt/data/elasticsearch-5.6.16 /opt/data/elasticsearch-6.8.23" -b
+    ```
+
+  12. With ES 6 the cluster name is not allowed in data path, so we move data directory up one level.
```

Just to be specific 😅 
```suggestion
  12. With Elasticsearch 6 the cluster name is not allowed in data path, so we move data directory up one level.
```